### DISCUSSION TITLE
False positive when set item for the dict which has auto deduced type?

### DISCUSSION THREAD

**Discussion by scruel** [microsoft/pyright#8652] [Q&A] [answered]
```python x.py
q = {'bool': {'must': []}}
q['query_parsed'] = False
```

$ pyright x.py
/home/scruel/x.py
  /home/scruel/x.py:2:1 - error: Argument of type "Literal[False]" cannot be assigned to parameter "value" of type "dict[str, list[Unknown]]" in function "__setitem__"
    "Literal[False]" is incompatible with "dict[str, list[Unknown]]" (reportArgumentType)
1 error, 0 warnings, 0 informations

**Comment by erictraut** [comment, marked as answer]
Pyright is working as designed here, so I don't consider this a bug.

The inferred type of the expression assigned to `q` is `dict[str, dict[str, list[Any]]]`. The second line attempts to set a value in this dict object that violates its type, hence the type error.

For details about pyright's default inference behaviors for dict expressions, refer to [this documentation](https://microsoft.github.io/pyright/#/type-inference?id=dictionary-expressions).

You can override the default inference behaviors by providing a type declaration for `q` in this case.

```python
from typing import Any

q: dict[str, Any] = {"bool": {"must": []}}
q["query_parsed"] = False
```
